system/Psr/Cache/Pool.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Psr\Cache;
+
+use CodeIgniter\Cache\CacheInterface;
+use CodeIgniter\I18n\Time;
+use Config\Cache;
+use Config\Services;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+final class Pool implements CacheItemPoolInterface
+{
+	/**
+	 * The adapter to use.
+	 *
+	 * @var CacheInterface
+	 */
+	private $adapter;
+
+	/**
+	 * Deferred Items to be saved.
+	 *
+	 * @var array<string, Item>
+	 */
+	private $deferred = [];
+
+	/**
+	 * Initializes the underlying adapter
+	 * from an existing instance or from the
+	 * Cache Service (with optional config).
+	 *
+	 * @param object|null $object
+	 *
+	 * @throws CacheArgumentException
+	 */
+	public function __construct($object = null)
+	{
+		if (is_null($object))
+		{
+			$this->adapter = Services::cache();
```

Comment:
I think standardizing on using the `service()` helper is a good thing for us. I think it keeps us safer if we make changes down the line to ensure proper service detection ordering since it doesn't call a specific class.